app/ui/src/app/integration/edit-page/action-configure/action-configure.component.ts

```diff
@@ -191,7 +191,7 @@ export class IntegrationConfigureActionComponent extends FlowPage
       this.loading = false;
       return;
     }
-    if (!descriptor || descriptor === undefined || !descriptor.propertyDefinitionSteps) {
+    if (!descriptor || descriptor === undefined || Object.keys(descriptor.propertyDefinitionSteps[0]).length === 0) {
```

Comment:
Isn't this going to cause an NPE if `propertyDefinitionSteps` is undefined?

Reply:
yes, but the way I have seen we are propagating logical empty values, it's never undefined.

somehow, when we have no property we represent that as a `[{}]`, that I'm happy to get rid of, but I was afraid to touch and risking of creating further issues